tests/test_core.py

```diff
@@ -0,0 +1,98 @@
+import ixmp
+import message_ix
+import pytest
+
+import pandas as pd
+import pandas.util.testing as pdt
+
+from message_ix import Scenario
+from numpy import testing as npt
+from testing_utils import test_mp
+
+
+msg_args = ('canning problem (MESSAGE scheme)', 'standard')
+
+
+def test_add_spatial_single(test_mp):
+    scen = Scenario(test_mp, *msg_args, version='new')
+    data = {'country': 'Austria'}
+    scen.add_spatial_sets(data)
+
+    exp = ['World', 'Austria']
+    obs = scen.set('node')
+    npt.assert_array_equal(obs, exp)
+
+    exp = ['World', 'global', 'country']
```

Comment:
FYI @danielhuppmann, I've confirmed that the `add_spatial_sets` only adds `country` and `state` in these tests. `World` and `global` come in automatically on the `ixmp` side.

Reply:
indeed, that's a bug on the Java source side, see https://github.com/iiasa/ixmp_source/issues/120